### PR TITLE
Fix409 e

### PR DESCRIPTION
**Pull request by alanlanglois**
## Detailed purpose of the PR

A new attempt to fix 409 error.
Splitting process in two different functions and adding a isSynchronising flag in the state in order to avoid multi update. 
Still need to :
- [x] check test-unit

### Testing strategy

- [x] I ensured my code is tested: Tested on iOS Simulator and Android emulator
- [ ] I took care to mention breaking changes and extra libs if appropriate

### CODE REVIEW

File: src/redux/actions/progressions/synchronize.test.js [hunk 1]
```diff
@@ -117,16 +144,15 @@ describe('Progressions synchronization', () => {
...
 
       expect(services.Progressions.updateSynchronizedProgressionIds).toHaveBeenCalledTimes(2);
       expect(services.Progressions.synchronize).toHaveBeenCalledTimes(2);
       expect(services.Progressions.updateSynchronizedProgressionIds).toHaveBeenCalledWith([
         successProgression._id,
         failureProgression._id
       ]);
-      expect(store.dispatch).toHaveBeenCalledTimes(2);
-      expect(actual).toBeUndefined();
+      expect(store.dispatch).toHaveBeenCalledTimes(4);
```
Comment (Fausto95) [inline src/redux/actions/progressions/synchronize.test.js:155]: It'd be nice to also test the number of times that `innerDispatch` should be called

File: src/redux/actions/progressions/synchronize.test.js [hunk 2]
```diff
@@ -152,27 +178,39 @@ describe('Progressions synchronization', () => {
...
+      await synchronizeProgressions(store.dispatch, store.getState, {services});
 
       expect(services.Progressions.updateSynchronizedProgressionIds).toHaveBeenCalledTimes(1);
       expect(services.Progressions.synchronize).toHaveBeenCalledTimes(0);
       expect(services.Progressions.updateSynchronizedProgressionIds).toHaveBeenCalledWith([
         successProgression._id
       ]);
-      expect(store.dispatch).toHaveBeenCalledTimes(1);
-      expect(actual).toBeUndefined();
+      expect(store.dispatch).toHaveBeenCalledTimes(3);
```
Comment (Fausto95) [inline src/redux/actions/progressions/synchronize.test.js:213]: Same as above, test how many times `innerDispatch` should be called

File: src/redux/actions/progressions/synchronize.js [hunk 3]
```diff
@@ -19,84 +19,126 @@ export const SYNCHRONIZE_FAILURE = '@@progression/SYNCHRONIZE_FAILURE';
...
-export const synchronizeProgressions: StoreAction<Action> = async (dispatch, getState, options) => {
+export const synchronizeRequest = (): Action => ({
+  type: SYNCHRONIZE_REQUEST
+});
+export const synchronizeSuccess = (): Action => ({
+  type: SYNCHRONIZE_SUCCESS
+});
+export const synchronizeFailure = (): Action => ({
+  type: SYNCHRONIZE_FAILURE
+});
```
Comment (VincentCATILLON) [inline src/redux/actions/progressions/synchronize.js:39]: these actions should take id in payload to track it :)

File: src/redux/utils/state-extract.js [hunk 4]
```diff
@@ -62,6 +62,9 @@ export const getToken = (state: StoreState): TokenState =>
 export const getBrand = (state: StoreState): BrandState =>
   state.authentication && state.authentication.brand;
 
+export const isProgressionsSynchronizing = (state: StoreState): boolean =>
+  state.progressions && state.progressions.isSynchronizing;
```
Comment (VincentCATILLON) [inline src/redux/utils/state-extract.js:66]: `state.progressions.isSynchronizing` is sufficient as the state can't be undefined or null

**Pull request merged**